### PR TITLE
POC map react native leaflet

### PR DESCRIPTION
**Pull request by camcamBoir**
POC pour la librairie react-native-leaflet (https://github.com/pavel-corsaghin/react-native-leaflet/tree/main):
- affichage sur la page des détails de l'alerte d'une carte centrée sur les coordonnées du feu
- affichage d'un cercle à l'endroit des coordonnées de la caméra
- affichage du triangle de vision dans lequel se trouve le feu selon l'azimuth de la caméra
image

### CODE REVIEW

File: PyroApp/services/alerts.service.ts [hunk 1]
```diff
@@ -13,6 +29,46 @@ async function getAlerts() {
   }
 }
 
+// Fonction pour convertir les degrés en radians
+function toRadians(degrees: number): number {
+  return degrees * (Math.PI / 180);
+}
+
+function calculateCoordinatesTriangle(alert: alert): number[][] {
```
Comment (aumeunier) [inline PyroApp/services/alerts.service.ts]: @MateoLostanlen : can you review this function please? 🙏

File: PyroApp/services/auth.service.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ async function login(username: string, password: string) {
 
     await saveToken(response.data.access_token);
   } catch (error) {
-    console.log(error);
+    throw error;
```
Comment (aumeunier) [inline PyroApp/services/auth.service.ts]: you don't need the try / catch in that case

File: PyroApp/source/Screens/MainScreen/MainScreen.js [hunk 3]
```diff
@@ -4,45 +4,93 @@ import {alertsService} from '../../../services/alerts.service';
 import CustomButton from '../../Components/CustomButton';
 import {authService} from '../../../services/auth.service';
 import {useNavigation} from '@react-navigation/native';
+import {LeafletView} from 'react-native-leaflet-view';
+import {SafeAreaView} from 'react-native-safe-area-context';
 
 const MainScreen = () => {
   const [alert, setAlert] = useState(undefined);
+  const [mapCenter, setMapCenter] = useState({
```
Comment (aumeunier) [inline PyroApp/source/Screens/MainScreen/MainScreen.js]: I would advice against those default values. It's better to not show the map or to have a default map center that is somewhere kinda useful (like France)
I've written quite a few apps with maps and you always end up with some weird cases where your map ends up being in the middle of the ocean :(

File: PyroApp/source/Screens/MainScreen/MainScreen.js [hunk 4]
```diff
@@ -4,45 +4,93 @@ import {alertsService} from '../../../services/alerts.service';
...
+    lat: 1.0,
+    lng: 1.0,
+  });
+  const [triangleCoordinates, setTriangleCoordinates] = useState([
```
Comment (aumeunier) [inline PyroApp/source/Screens/MainScreen/MainScreen.js]: This should be nullable / undefined
Also, you could type it

File: PyroApp/source/Screens/MainScreen/MainScreen.js [hunk 5]
```diff
@@ -4,45 +4,93 @@ import {alertsService} from '../../../services/alerts.service';
 import CustomButton from '../../Components/CustomButton';
 import {authService} from '../../../services/auth.service';
 import {useNavigation} from '@react-navigation/native';
+import {LeafletView} from 'react-native-leaflet-view';
+import {SafeAreaView} from 'react-native-safe-area-context';
 
 const MainScreen = () => {
   const [alert, setAlert] = useState(undefined);
```
Comment (aumeunier) [inline PyroApp/source/Screens/MainScreen/MainScreen.js]: Typescript doesn't quite like useState without types if you have an undefined

File: PyroApp/source/Screens/MainScreen/MainScreen.js [hunk 6]
```diff
@@ -4,45 +4,93 @@ import {alertsService} from '../../../services/alerts.service';
 import CustomButton from '../../Components/CustomButton';
 import {authService} from '../../../services/auth.service';
 import {useNavigation} from '@react-navigation/native';
+import {LeafletView} from 'react-native-leaflet-view';
+import {SafeAreaView} from 'react-native-safe-area-context';
 
 const MainScreen = () => {
```
Comment (aumeunier) [inline PyroApp/source/Screens/MainScreen/MainScreen.js]: Looks like you're cheating, the file is a .js one !

File: PyroApp/source/Screens/MainScreen/MainScreen.js [hunk 7]
```diff
@@ -4,45 +4,93 @@ import {alertsService} from '../../../services/alerts.service';
...
     await authService.removeToken();
+    setAlert(undefined);
     navigation.navigate('Signin');
   }
 
   useEffect(() => {
     const fetchData = async () => {
       try {
-        const res = await alertsService.getAlerts();
+        const res = await alertsService.getAlert();
```
Comment (aumeunier) [inline PyroApp/source/Screens/MainScreen/MainScreen.js]: can this be undefined ?

**Review by aumeunier — CHANGES_REQUESTED** [review]

File: PyroApp/source/Screens/MainScreen/MainScreen.js (hunk 3 above)
Comment (camcamBoir) [reply PyroApp/source/Screens/MainScreen/MainScreen.js]: j'ai mis une localisation en France du coup, dans la région où ils veulent détecter des incendies

File: PyroApp/source/Screens/MainScreen/MainScreen.js (hunk 7 above)
Comment (camcamBoir) [reply PyroApp/source/Screens/MainScreen/MainScreen.js]: nope, j'ai corrigé

File: PyroApp/source/Screens/MainScreen/MainScreen.js (hunk 6 above)
Comment (camcamBoir) [reply PyroApp/source/Screens/MainScreen/MainScreen.js]: haha oui @timogrand avait créé des fichiers js

**Review by aumeunier — APPROVED** [review]

File: PyroApp/services/alerts.service.ts (hunk 1 above)
Comment (MateoLostanlen) [reply PyroApp/services/alerts.service.ts]: LGTM

**Comment by MateoLostanlen** [comment]
Hi @camcamBoir ,

thank you very much for this PR. It seems ok to me. Only one thing: try to put the comments in English. The code must be usable by everyone.

**Pull request merged**